apps/ex_nvr_web/lib/ex_nvr_web/controllers/api/event_controller.ex

```diff
@@ -0,0 +1,79 @@
+defmodule ExNVRWeb.API.EventController do
+  use ExNVRWeb, :controller
+
+  action_fallback ExNVRWeb.API.FallbackController
+  plug(:extract_device when action in [:create, :index])
+  plug(:ensure_type when action in [:create, :index])
+
+  alias Plug.Conn
+  alias ExNVR.{Events, Devices}
+
+  @spec create(Conn.t(), map()) :: {:error, Ecto.Changeset.t()} | Conn.t()
+  def create(%Conn{body_params: params} = conn, _) do
+    device = conn.assigns.device
+    type = conn.assigns.type
+
+    with {:ok, event} <- Events.create(params, device, type),
+         event <- format_event(event, device) do
+      conn
+      |> put_status(201)
+      |> json(%{event: event})
+    end
+  end
+
+  @spec index(Conn.t(), map()) :: Conn.t() | {:error, Ecto.Changeset.t()}
+  def index(%Conn{} = conn, _params) do
+    device = conn.assigns.device
+    type = conn.assigns.type
+
+    events =
+      Events.list(device.id, type)
+      |> Enum.map(&format_event(&1, device))
```

Comment:
For this endpoint, you can just use the authentication plug we already have and then get all events, use filter to get specific events (they're optional). Use pagination, sorting and filtering like recordings endpoint (use Flop for that).